databricks/koalas/groupby.py

```diff
@@ -2155,6 +2155,73 @@ def expanding(self, min_periods=1):
         """
         return ExpandingGroupby(self, min_periods=min_periods)
 
+    def get_group(self, name):
+        """
+        Construct DataFrame from group with provided name.
+
+        Parameters
+        ----------
+        name : object
+            The name of the group to get as a DataFrame.
+
+        Returns
+        -------
+        group : same type as obj
+
+        Examples
+        --------
+        >>> kdf = ks.DataFrame([('falcon', 'bird', 389.0),
+        ...                     ('parrot', 'bird', 24.0),
+        ...                     ('lion', 'mammal', 80.5),
+        ...                     ('monkey', 'mammal', np.nan)],
+        ...                    columns=['name', 'class', 'max_speed'],
+        ...                    index=[0, 2, 3, 1])
+        >>> kdf
+             name   class  max_speed
+        0  falcon    bird      389.0
+        2  parrot    bird       24.0
+        3    lion  mammal       80.5
+        1  monkey  mammal        NaN
+
+        >>> kdf.groupby("class").get_group("bird").sort_index()
+             name class  max_speed
+        0  falcon  bird      389.0
+        2  parrot  bird       24.0
+
+        >>> kdf.groupby("class").get_group("mammal").sort_index()
+             name   class  max_speed
+        1  monkey  mammal        NaN
+        3    lion  mammal       80.5
+        """
+        groupkeys = self._groupkeys
+        if not is_hashable(name):
+            raise TypeError("unhashable type: '{}'".format(type(name).__name__))
+        elif len(groupkeys) > 1 and not isinstance(name, tuple):
+            raise ValueError("must supply a tuple to get_group with multiple grouping keys")
+        if isinstance(name, str):
+            name = [name]
+        cond = F.lit(True)
+        for groupkey, item in zip(groupkeys, name):
+            scol = groupkey.spark.column
+            cond = cond & (scol == item)
+        internal = self._kdf._internal.with_filter(cond)
+        if internal.spark_frame.head() is None:
+            raise KeyError(name)
+        if self._agg_columns_selected:
+            internal = self._kdf._internal
```

Comment:
Why start from `self._kdf._internal` again?

Reply:
I think maybe this `internal` has different identity from `internal` above since above one is copied when performing `with_filter` ??

without this, cannot find the proper Spark column as below.

```python
pyspark.sql.utils.AnalysisException: Resolved attribute(s) max_speed#3 missing from __index_level_0__#0L,max_speed#199,__natural_order__#207L in operator !Project [__index_level_0__#0L, (max_speed#3 + cast(1 as double)) AS max_speed#199]. Attribute(s) with the same name appear in the operation: max_speed. Please check if the right attribute(s) are used.;;
!Project [__index_level_0__#0L, (max_speed#3 + cast(1 as double)) AS max_speed#199]
+- Project [__index_level_0__#0L, max_speed#199, monotonically_increasing_id() AS __natural_order__#207L]
   +- Project [__index_level_0__#0L, (max_speed#3 + cast(1 as double)) AS max_speed#199]
      +- Filter (true AND (class#2 = mammal))
         +- Project [__index_level_0__#0L, name#1, class#2, max_speed#3, monotonically_increasing_id() AS __natural_order__#8L]
            +- LogicalRDD [__index_level_0__#0L, name#1, class#2, max_speed#3], false
```

Reply:
Then the flow is weird. It should be:

```py
cond = F.lit(True)
for groupkey, item in zip(groupkeys, name):
    ...
    cond = ...

if self._agg_columns_selected:
    ...
else:
    internal = self._kdf._internal.with_filter(cond)

if internal.spark_frame.head() is None:
    ...

...
```

Reply:
Oops.. totally agree. It was my bad 🙀